kornia/augmentation/utils/param_validation.py

```diff
@@ -94,83 +96,78 @@ def _singular_range_check(
 
 
 def _tuple_range_reader(
-    input_range: Union[torch.Tensor, float, tuple],
+    input_range: Union[Tensor, float, tuple],
     target_size: int,
     device: Optional[torch.device] = None,
     dtype: Optional[torch.dtype] = None,
-) -> torch.Tensor:
+) -> Tensor:
     """Given target_size, it will generate the corresponding (target_size, 2) range tensor for element-wise params.
 
     Example:
-    >>> degree = torch.tensor([0.2, 0.3])
+    >>> degree = tensor([0.2, 0.3])
     >>> _tuple_range_reader(degree, 3)  # read degree for yaw, pitch and roll.
     tensor([[0.2000, 0.3000],
             [0.2000, 0.3000],
             [0.2000, 0.3000]])
     """
     target_shape = torch.Size([target_size, 2])
-    if not torch.is_tensor(input_range):
-        if isinstance(input_range, (float, int)):
+
+    if isinstance(input_range, Tensor):
+        # TODO: Investigate if we should keeping ignoring the `device` argument when `input_range` is a Tensor
```

Comment:
Just to point out here, the problem was with this parameter validation function, which was not considering the `dtype` argument when the input is a tensor. Currently, it continues to ignore the `device` parameter if it is a tensor, otherwise it causes `Expected all tensors to be on the same device`

Reply:
Maybe the right thing here, should be considering the device argument, and after generate the parameter move to the desired device?